### PR TITLE
11573 SP P&R validation, etc

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#11573

*Description of changes:*
- changed Add Edit components to section + rows/cols + articles
- implemented form validation and error styling
- updated max org name length to 150
- added error checkbox icon style override
- fixed Add Edit form validation error
- updated some private/protected
- used shared DocumentDelivery component instead of local one
- updated HelpBusinessNumber for future reuse
- updated "edit business" label
- commented out lookup toggle link
- set step invalid when adding/editing
- fixed event-prop looping with BaseAddress
- fixed form validation logic
- misc cleanup
- app version = 3.5.7

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Registration/RegAddEditOrgPerson.vue [hunk 1]
```diff
@@ -5,270 +5,279 @@
...
-                </p>
-              </v-card>
-
-              <v-form
-                ref="addPersonOrgForm"
-                class="appoint-form"
-                :class="{ 'mt-8': isCompletingParty && !isRoleStaff}"
-                v-model="addPersonOrgFormValid"
-                v-on:submit.prevent
+    <section class="px-6 py-10" :class="{ 'invalid-section': !addPersonOrgFormValid }">
```
Comment (severinbeauvais) [inline src/components/Registration/RegAddEditOrgPerson.vue:8]: Please hide whitespace (⚙️ icon at top of Files Changed page) to see what actually changed.

File: src/components/Registration/RegPeopleAndRoles.vue [hunk 2]
```diff
@@ -105,22 +105,19 @@
...
-        :activeIndex="activeIndex"
-        :existingCompletingParty="completingParty"
-        @addEditPerson="onAddEditPerson($event)"
-        @removePerson="onRemovePerson($event)"
-        @resetEvent="resetData()"
-        @removeCompletingPartyRole="onRemoveCompletingPartyRole()"
-      />
-    </v-card>
+    <v-expand-transition>
+      <v-card flat v-if="showOrgPersonForm" class="mt-8">
```
Comment (severinbeauvais) [inline src/components/Registration/RegPeopleAndRoles.vue:109]: Transitions must wrap a code block that is conditionally rendered (v-if) or displayed (v-show). So this transition did nothing in the RegAddEditOrgPerson component itself and needs to be here instead.

File: src/components/common/AddEditOrgPerson.vue [hunk 3]
```diff
@@ -5,208 +5,219 @@
...
-                    <!-- NB: only staff can change Completing Party names -->
-                    <v-text-field
-                      filled
-                      class="item"
-                      label="First Name"
-                      id="person__first-name"
-                      v-model.trim="orgPerson.officer.firstName"
-                      :rules="Rules.FirstNameRules"
-                      :readonly="isCompletingParty && !isRoleStaff && isTypeCoop"
+    <section class="px-6 py-10" :class="{ 'invalid-section': !addPersonOrgFormValid }">
```
Comment (severinbeauvais) [inline src/components/common/AddEditOrgPerson.vue:8]: Same changes as RegAddEditOrgPerson for consistency.

File: src/components/Registration/RegAddEditOrgPerson.vue [hunk 4]
```diff
@@ -5,270 +5,288 @@
...
+                ref="mailingAddressNew"
+                class="mt-4 mb-n6"
+                :editing="true"
+                :schema="PersonAddressSchema"
+                :address="inProgressMailingAddress"
+                @update:address="updateMailingAddress($event)"
+                @valid="updateMailingAddressValidity($event)"
+              />
+              <!-- dummy component to make form invalid if mailing address is invalid -->
+              <v-input class="d-none" :rules="[() => mailingAddressValid]" />
```
Comment (severinbeauvais) [inline src/components/Registration/RegAddEditOrgPerson.vue:239]: I tried using `v-validate` and `vuelidate` to do this, and this workaround ended up being the simplest (and the one that worked).

I created [11737](https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/bcgov/entity/11737) to update BaseAddress and no longer need this workaround.

File: src/mixins/add-edit-org-person-mixin.ts [hunk 5]
```diff
@@ -222,23 +225,35 @@ export default class AddEditOrgPersonMixin extends Vue {
   }
 
   // Event Handlers
-  private updateMailingAddress (val): void {
-    this.inProgressMailingAddress = val
+  protected updateMailingAddress (address: AddressIF): void {
+    // only update if equal
+    // (workaround for BaseAddress always telling us there's a new address)
```
Comment (severinbeauvais) [inline src/mixins/add-edit-org-person-mixin.ts:230]: ... which is due to the prop-event looping issue

File: src/views/DissolutionFirm/DissolutionFirm.vue [hunk 6]
```diff
@@ -77,10 +77,10 @@
         :documentOptionalEmail="getDocumentDelivery.documentOptionalEmail"
         @update:optionalEmail="setDocumentOptionalEmail($event)"
         @valid="setDocumentOptionalEmailValidity($event)"
-        additionalLabel="partners"
+        additionalLabel="Partners"
         contactLabel="Business Contact"
       />
-      <!-- TODO change: additionalEmail & additionalLabel are sample data-->
+      <!-- SB TODO: update additionalLabel & additionalLabel (sample data)-->
```
Comment (severinbeauvais) [inline src/views/DissolutionFirm/DissolutionFirm.vue:83]: SB = "Shabeeb" (not Severin)

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by cameron-freshworks — APPROVED** [review]
Looks great!

**Pull request merged**